tests/Doctrine/Tests/ORM/Query/SelectSqlGenerationTest.php

```diff
@@ -2314,6 +2314,18 @@ public function testHavingRegressionUsingVariableWithMathOperatorsExpression($op
         );
     }
 
+    /**
+     * GitHub issue #7846: https://github.com/doctrine/orm/issues/7846
+     * @group 7846
+     */
+    public function testAliasCanBeUsedWithFunctions()
+    {
+        $this->assertSqlGeneration(
+            "SELECT u.name AS foo FROM Doctrine\Tests\Models\CMS\CmsUser u WHERE LOWER(foo) = '3'",
```

Comment:
What happens with `LOWER(u.name) = '3'`?

Reply:
It works, generated SQL is `SELECT c0_.name AS name_0 FROM cms_users c0_ WHERE LOWER(c0_.name_0) = '3'`
BTW there's a typo in my test, I will be fixing it.